### PR TITLE
fortran-csv-module: update to 2.1.0; fpm/fortran PG: use gcc13 until next release

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

No revbump needed for any fortran packages, nothing changed for them.

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: _resources/port1.0/group/fortran-1.0.tcl [hunk 1]
```diff
@@ -21,6 +21,15 @@ if {${os.major} < 14} {
 	git.cmd                 ${prefix}/bin/git
 }
 
+# https://github.com/fortran-lang/fpm/issues/1059
+# Drop this upon the next release of FPM:
+compiler.blacklist-append \
+                    macports-gcc-14
+if {${os.platform} ne "darwin" || ${os.major} > 9} {
```
Comment (reneeotten) [inline _resources/port1.0/group/fortran-1.0.tcl:28]: the os.major check should go together with the os.platform check for darwin; this isn't correct anymore now (I do understand the change, but you should include os.platform after the OR statement)

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: _resources/port1.0/group/fortran-1.0.tcl (hunk 1 above)
Comment (barracuda156) [reply _resources/port1.0/group/fortran-1.0.tcl:28]: Someone suggested me this form (maybe @nilason but not sure), and it looked logical: once not-darwin is filtered out, what is left is surely darwin only. Just to understand the issue, why won’t it work?

I will make the change, of course, no argument on that.

Comment (reneeotten) [reply _resources/port1.0/group/fortran-1.0.tcl:28]: ah, I did misread that - should be good indeed

Comment (barracuda156) [reply _resources/port1.0/group/fortran-1.0.tcl:28]: Ok, then I leave it as is.

**Pull request merged**